Reference/Notifications/Creating-And-Publishing-Notifications.md

```diff
@@ -0,0 +1,115 @@
+
+## Example
+
+```C#
+using System;
+using System.Threading.Tasks;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Scoping;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Infrastructure.HostedServices;
+using Umbraco.Cms.Web.UI.Notifications;
+
+namespace Umbraco.Cms.Web.UI
+{
+    public class CleanUpYourRoom : RecurringHostedServiceBase
+    {
+        private readonly IContentService _contentService;
+        private readonly IScopeProvider _scopeProvider;
+        private readonly IEventAggregator _eventAggregator;
+        private static TimeSpan HowOftenWeRepeat => TimeSpan.FromMinutes(5);
+        private static TimeSpan DelayBeforeWeStart => TimeSpan.FromMinutes(1);
+
+        public CleanUpYourRoom(
+            IContentService contentService,
+            IScopeProvider scopeProvider,
+            IEventAggregator eventAggregator)
+            : base(HowOftenWeRepeat, DelayBeforeWeStart)
+        {
+            _contentService = contentService;
+            _scopeProvider = scopeProvider;
+            _eventAggregator = eventAggregator;
+        }
+
+        public override Task PerformExecuteAsync(object state)
+        {
+            // This will be published immediately
+            _eventAggregator.Publish(new CleanYourRoomStartedNotification());
+
+            using IScope scope = _scopeProvider.CreateScope();
+            int numberOfThingsInBin = _contentService.CountChildren(Constants.System.RecycleBinContent);
+
+            if (_contentService.RecycleBinSmells())
+            {
+                _contentService.EmptyRecycleBin(userId: -1);
+                // This will only be published when the scope is completed and disposed.
+                scope.Notifications.Publish(new RoomCleanedNotification(numberOfThingsInBin));
+            }
+
+            // Remember to complete the scope when done.
+            scope.Complete();
+            return Task.CompletedTask;
+        }
+    }
+}
+```
+
+In this case the `CleanYourRoomStartedNotification` will always be published imidiately, however `RoomCleanedNotification` will only be published once the operation is done, and if you remove the `scope.Complete();` line it will never be published, the recycle bin won't be emptied either. 
```

Comment:
```suggestion
In this case, the `CleanYourRoomStartedNotification` will always be published immediately, however, `RoomCleanedNotification` will only be published once the operation is done, and if you remove the `scope.Complete();` line it will never be published, the recycle bin won't be emptied either. 
```